packages/lib-classifier/src/store/SubjectStore.spec.js

```diff
@@ -1,59 +1,62 @@
+
+const clientStub = stubPanoptesJs({ subjects, workflows: workflow })
+
+describe.only('Model > SubjectStore', function () {
+  function setupStores(panoptesClientStub = clientStub) {
+    const store = RootStore.create({
+      classifications: {},
+      dataVisAnnotating: {},
+      drawing: {},
+      feedback: {},
+      fieldGuide: {},
+      subjectViewer: {},
+      tutorials: {},
+      workflowSteps: {},
+      userProjectPreferences: {}
+    }, {
+        client: panoptesClientStub,
+        authClient: { checkBearerToken: () => Promise.resolve(), checkCurrent: () => Promise.resolve() }
+    })
+    sinon.spy(store.subjects, 'populateQueue')
 
-const clientStub = {
-  panoptes: {
-    get () {
-      return Promise.resolve({
-        body: {
-          subjects
-        }
-      })
-    }
-  }
-}
-
-describe('Model > SubjectStore', function () {
-  function setupStores (rootStore) {
-    sinon.stub(rootStore.classifications, 'createClassification')
-    rootStore.projects.setResource(project)
-    rootStore.workflows.setResource(workflow)
-    rootStore.workflows.setActive(workflow.id)
+    store.projects.setResource(project)
+    store.projects.setActive(project.id)
+    store.workflows.setResource(workflow)
+    store.workflows.setActive(workflow.id)
+    return store
   }
   describe('Actions > advance', function () {
+    let rootStore
     before(function () {
-      rootStore = RootStore.create(
-        { projects: ProjectStore.create(), subjects: SubjectStore.create(), workflows: WorkflowStore.create() },
-        { client: clientStub }
-      )
+      rootStore = setupStores()
     })
 
-    it('should make the next subject in the queue active when calling `advance()`', function (done) {
-      setupStores(rootStore)
-      rootStore.subjects.populateQueue().then(() => {
-        expect(rootStore.subjects.active.id).to.equal(rootStore.subjects.resources.values().next().value.id)
-        rootStore.subjects.advance()
-        expect(rootStore.subjects.active.id).to.equal(rootStore.subjects.resources.values().next().value.id)
-        expect(rootStore.subjects.resources.get('1')).to.be.undefined()
-      }).then(done, done)
+    it('should make the next subject in the queue active when calling `advance()`', function () {
+      expect(rootStore.subjects.active.id).to.equal(rootStore.subjects.resources.values().next().value.id)
+      rootStore.subjects.advance()
+      expect(rootStore.subjects.active.id).to.equal(rootStore.subjects.resources.values().next().value.id)
+      expect(rootStore.subjects.resources.get('1')).to.be.undefined()
     })
 
     describe('with less than three subjects in the queue', function () {
-      let populateSpy
+      let rootStore
 
       before(function () {
-        populateSpy = sinon.spy(rootStore.subjects, 'populateQueue')
+        rootStore = setupStores()
+      })
+
+      beforeEach(function () {
         while (rootStore.subjects.resources.size > 2) {
```

Comment:
This will not run before each test unless the queue is reset to 3 or more subjects. Otherwise, it only runs before the first test, then the queue size is 2 for subsequent tests.

Reply:
It was only applicable to the first test in this block, so I'll move the while loop into that test to make sure it's clear.